db/migrate/20240205155801_add_discarded_at_to_users.rb

```diff
@@ -0,0 +1,6 @@
+class AddDiscardedAtToUsers < ActiveRecord::Migration[7.1]
+  def change
+    add_column :users, :discarded_at, :datetime
+    add_index :users, :discarded_at
```

Comment:
I wonder if we need a composite index for

```ruby
add_index :users, [:type, :discarded_at]
```

Reply:
Furthermore, it _might_ even be beneficial to do:

```ruby
add_index :users, [:type, :discarded_at, :email]
```

We'd need to check our queries though.

Reply:
resolved in[b1e480f](https://github.com/DFE-Digital/itt-mentor-services/pull/222/commits/b1e480fe5463bb30ad62e8d24c407cc6fb3cb962)